google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/scaladsl/BigQueryJobs.scala

```diff
@@ -68,21 +65,16 @@ private[scaladsl] trait BigQueryJobs { this: BigQueryRest =>
   def cancelJob(
       jobId: String,
       location: Option[String] = None
-  )(implicit system: ClassicActorSystemProvider, settings: BigQuerySettings): Future[JobCancelResponse] = {
+  )(implicit system: ClassicActorSystemProvider, settings: GoogleSettings): Future[JobCancelResponse] = {
     import BigQueryException._
     import SprayJsonSupport._
-    implicit val ec = system.classicSystem.dispatcher
     val uri = BigQueryEndpoints.jobCancel(settings.projectId, jobId)
     val query = ("location" -> location) ?+: Query.Empty
-    BigQueryHttp()
-      .retryRequestWithOAuth(HttpRequest(POST, uri.withQuery(query)))
-      .flatMap { response =>
-        Unmarshal(response.entity).to[JobCancelResponse]
-      }(system.classicSystem.dispatcher)
+    singleRequest[JobCancelResponse](HttpRequest(POST, uri.withQuery(query)))
```

Comment:
Yes, sorry, maybe this became too convoluted.

1. Inside `GoogleHttp` I consolidated `singleRequest[WithOAuth]` with `retryRequest[WithOAuth]` to `single[Authenticated]Request` which retries if a `Retry(ex)` is thrown.
2. In the public API `Google` I expose `singleRequest` which just calls `singleAuthenticatedRequest`.
3. I mixed `Google` into `BigQuery` and preferred calls to public `Google` APIs instead of directly to internal `GoogleHttp`

Reply:
Got it.